scanner/indexer/indexer.go

```diff
@@ -140,6 +146,19 @@ func (i *indexerImpl) IndexContainerImage(
 	return i.indexer.Index(ctx, manifest)
 }
 
+func getLayerHTTPClient(ctx context.Context, imgRef name.Reference, auth authn.Authenticator, timeout time.Duration) (*http.Client, error) {
+	reg := imgRef.Context().Registry
+	tr, err := transport.NewWithContext(ctx, reg, auth, http.DefaultTransport, nil)
```

Comment:
`http.DefaultTransportation` is probably fine, but there is also a [`remote.DefaultTransportation`](https://github.com/google/go-containerregistry/blob/v0.16.1/pkg/v1/remote/options.go#L113) which is based on the `http` one.

Reply:
Also, let's wrap the chosen `http.RoundTripper` with `transport.NewRetry()`

Reply:
Let's also make it clear we just want pull scope. See https://github.com/quay/clair/blob/v4.7.1/cmd/clairctl/client.go#L52 for more info

Reply:
Oh, good. Doing it.